From a64ab8564b754274b25e9602056ce6b1f5e4dc39 Mon Sep 17 00:00:00 2001
From: Marek 'marx' Grac <mgrac@redhat.com>
Date: Wed, 18 Feb 2015 10:22:41 +0100
Subject: [PATCH 08/16] Revert "COMPATIBILITY BREAK: fence_rsb replace -n /
 telnet_port with standard --ipport / ipport"

This reverts commit dbc406ad1c5a2f981c04cd6b6d7ebc02aab1eaf5.
---
 fence/agents/rsb/fence_rsb.py | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/fence/agents/rsb/fence_rsb.py b/fence/agents/rsb/fence_rsb.py
index 2ee85fe..4a7e246 100755
--- a/fence/agents/rsb/fence_rsb.py
+++ b/fence/agents/rsb/fence_rsb.py
@@ -39,13 +39,24 @@ def set_power_status(conn, options):
 	conn.log_expect(options["--command-prompt"], int(options["--shell-timeout"]))
 
 def main():
-	device_opt = ["ipaddr", "login", "passwd", "secure", "cmd_prompt", "telnet"]
+	device_opt = ["ipaddr", "telnet_port", "login", "passwd", "secure", "cmd_prompt", "telnet"]
 
 	atexit.register(atexit_handler)
 
+	all_opt["telnet_port"] = {
+		"getopt" : "n:",
+                "longopt" : "telnet_port",
+                "help" : "-n                             TCP port to use (deprecated, use -u)",
+                "required" : "0",
+                "shortdesc" : "TCP port to use for connection with device (default is 3172 for telnet)",
+                "order" : 1
+	}
 	all_opt["cmd_prompt"]["default"] = ["to quit:"]
 
 	opt = process_input(device_opt)
+	# option -n for backward compatibility (-n is normally port no)
+	if 1 == opt.has_key("--telnet_port"):
+		opt["--ipport"] = opt["--telnet_port"]
 
 	if not opt.has_key("--ssh") and not opt.has_key("--ipport"):
 		# set default value like it should be set as usually
-- 
1.9.3

